controllers/harborconfiguration_controller.go

```diff
@@ -391,20 +391,52 @@ func getConcreteHarborType(ctx context.Context, crdClient dynamic.ResourceInterf
 
 func deleteAll(ctx context.Context, harborConfiguration harborconfigurationv1alpha1.HarborConfiguration, client *apiv2.RESTClient) (ctrl.Result, error) {
 	deletionLog := ctrl.Log.WithName("deletion")
+	replicationFound, err := client.GetReplicationPolicyByName(ctx, harborConfiguration.Spec.Replication.Name)
+	if errors.Is(err, &harborerrors.ErrNotFound{}) {
+		deletionLog.Error(err, "skipping as policy does not exist.")
+	} else if err != nil {
+		deletionLog.Error(err, "failed to get replication policy by id to delete.")
```

Comment:
Should we return `err` in this case?

Reply:
The idea of not using return was so that the rest of the code will still execute and have a chance to delete (the project) even if the replication rule delete fails. I'm going to refactor the deletes into their own methods and I could change them to return, instead of log, errors if you'd like.

Reply:
If deletion of the replication rule fails the registry won't delete anyway as it is required to not be in use to be deleted. So the above logic only really applies for trying to delete a project.

Reply:
Can be merged with the failed CI for now until we have sorted that.